opm/parser/eclipse/EclipseState/Tables/SgofTable.hpp

```diff
@@ -21,58 +21,48 @@
 
 #include "SimpleTable.hpp"
 
+#include <opm/parser/eclipse/EclipseState/Tables/TableEnums.hpp>
+
 namespace Opm {
     // forward declaration
-    class TableManager;
-
     class SgofTable : public SimpleTable {
-        friend class TableManager;
 
-        /*!
-         * \brief Read the SGOF keyword and provide some convenience
-         *        methods for it.
-         */
-        void init(Opm::DeckItemConstPtr item)
+    public:
+        SgofTable(Opm::DeckItemConstPtr item)
         {
-            SimpleTable::init(item,
-                              std::vector<std::string>{"SG", "KRG", "KROG", "PCOG"});
+            m_schema = std::make_shared<TableSchema>();
+            m_schema->addColumn( ColumnSchema("SG"   , Table::STRICTLY_INCREASING , Table::DEFAULT_NONE));
+            m_schema->addColumn( ColumnSchema("KRG"  , Table::RANDOM              , Table::DEFAULT_LINEAR ));
+            m_schema->addColumn( ColumnSchema("KROG" , Table::RANDOM              , Table::DEFAULT_LINEAR ));
+            m_schema->addColumn( ColumnSchema("PCOG" , Table::RANDOM              , Table::DEFAULT_LINEAR ));
 
-            SimpleTable::checkNonDefaultable("SG");
-            SimpleTable::checkMonotonic("SG", /*isAscending=*/true);
-            SimpleTable::applyDefaultsLinear("KRG");
-            SimpleTable::applyDefaultsLinear("KROG");
-            SimpleTable::applyDefaultsLinear("PCOG");
+            SimpleTable::init( item );
         }
 
-    public:
-        SgofTable() = default;
-
-#ifdef BOOST_TEST_MODULE
-        // DO NOT TRY TO CALL THIS METHOD! it is only for the unit tests!
-        void initFORUNITTESTONLY(Opm::DeckItemConstPtr item)
-        { init(item); }
-#endif
-
-        using SimpleTable::numTables;
-        using SimpleTable::numRows;
-        using SimpleTable::numColumns;
-        using SimpleTable::evaluate;
-
-        const std::vector<double> &getSgColumn() const
-        { return SimpleTable::getColumn(0); }
+        const TableColumn& getSgColumn() const
+        {
+            return SimpleTable::getColumn(0);
+        }
 
-        const std::vector<double> &getKrgColumn() const
-        { return SimpleTable::getColumn(1); }
+        const TableColumn& getKrgColumn() const
+        {
```

Comment:
third style of curly braces for method bodies. I don't care much about it, though.